src/plugins/dashboard/server/saved_objects/migrate_match_all_query.ts

```diff
@@ -31,6 +37,7 @@ export const migrateMatchAllQuery: SavedObjectMigrationFn<any, any> = (doc) => {
       searchSource = JSON.parse(searchSourceJSON);
     } catch (e) {
       // Let it go, the data is invalid and we'll leave it as is
+      return doc;
```

Comment:
Can we add a test for all migration functions to verify the behaviour of `searchSourceJSON` being invalid JSON?